sklearn/cluster/tests/test_affinity_propagation.py

```diff
@@ -197,17 +205,17 @@ def test_affinity_propagation_non_convergence_regressiontest():
     assert_array_equal(np.array([0, 0, 0]), af.labels_)
 
 
-def test_equal_similarities_and_preferences():
+def test_equal_similarities_and_preferences(global_dtype):
```

Comment:
The previous test (`test_affinity_propagation_non_convergence_regressiontest`) should raise the warning for the different dtypes I think?

Reply:
and `test_affinity_propagation_poredict_non_convergence` as well.

Reply:
> The previous test (test_affinity_propagation_non_convergence_regressiontest) should raise the warning for the different dtypes I think?

Which difference of dtypes are you referring to?